R/duplicates.R

```diff
@@ -21,26 +21,30 @@ duplicates <- function(.data, ..., .keep_all = TRUE) {
   chk_flag(.keep_all)
 
   col <- rlang::ensyms(...)
-  if (length(col) == 0) {
-    col_names <- colnames(.data)
-  } else {
-    col_names <- vapply(col, rlang::as_string, character(1))
-  }
+  if (length(col) == 0) { col_names <- colnames(.data) }
+  else { col_names <- vapply(col, rlang::as_string, character(1)) }
   col_names <- unique(col_names)
   chk_vector(col_names)
   check_values(col_names, "")
   check_names(.data, col_names)
 
-  if (!length(col_names)) {
-    return(.data)
-  }
+  if (!length(col_names)) { return(.data) }
+  
+  grouped <- dplyr::is_grouped_df(.data)
+  groups <- dplyr::group_vars(.data)
+  groups_sym <- rlang::syms(groups)
+  
+  is_sf = any(class(.data) == "sf")
+  
+  .data <- tibble::as_tibble(.data)
+  
   .data_dup <- dplyr::select(.data, dplyr::all_of(col_names))
   .data_dup <- .data_dup[duplicated(.data_dup), , drop = FALSE]
   .data_dup <- unique(.data_dup)
-  .data <- dplyr::inner_join(.data, .data_dup, by = col_names)
-  if (!(.keep_all)) {
-    .data <- dplyr::select(.data, dplyr::all_of(col_names))
-  }
-  .data <- dplyr::as_tibble(.data)
+  
+  .data <- dplyr::inner_join(.data, .data_dup, by = col_names) 
+  if (!(.keep_all)) { .data <- dplyr::select(.data, dplyr::all_of(col_names)) }
+  if (grouped) { .data <- dplyr::group_by(.data, !!!groups_sym) }
+  if (is_sf) { .data <- poisspatial::ps_activate_sfc(.data) }
```

Comment:
I am not sure if we should set the active geometry here as it depends on the column name being "geometry".

Reply:
Yes I think we should the following functions to get, save and then set the active geometry name

https://r-spatial.github.io/sf/reference/st_geometry.html